src/lib.rs

```diff
@@ -496,7 +496,7 @@ impl MetadataCommand {
     }
 
     /// Runs configured `cargo metadata` and returns parsed `Metadata`.
-    pub fn exec(&mut self) -> Result<Metadata> {
+    pub fn exec(&self) -> Result<Metadata> {
```

Comment:
I understand the motiviation for this, but if it's not strictly necessary I'd like to withold it until the next major version bump. Please remove it from the PR and open an issue for it.